src/uct/ib/base/ib_iface.c

```diff
@@ -695,18 +712,44 @@ uct_ib_iface_roce_is_reachable(const uct_ib_device_gid_info_t *local_gid_info,
                                             &local_gid_info->gid,
                                             &sa_local) != UCS_OK)) {
         uct_iface_fill_info_str_buf(
-               params, "Couldn't convert local RoCE address to socket address");
+               params, "couldn't convert local RoCE address to socket address");
         return 0;
     }
 
     if (uct_ib_device_roce_gid_to_sockaddr(remote_ib_addr_af,
                                            remote_ib_addr + 1,
                                            &sa_remote) != UCS_OK) {
         uct_iface_fill_info_str_buf(
-               params, "Couldn't convert remote RoCE address to socket address");
+               params, "couldn't convert remote RoCE address to socket address");
+        return 0;
+    }
+
+    status = uct_ib_device_get_roce_ndev_name(dev, port_num,
```

Comment:
the function uct_ib_iface_roce_is_reachable is already quite big, maybe split it to sub-function, so that each one handles a different mode of reachability?